spec/models/provider_spec.rb

```diff
@@ -61,4 +61,38 @@
       expect(tenant.providers).to include(provider)
     end
   end
+
+  context "#orchestrate_destroy" do
+    let(:manager) { FactoryGirl.create(:ext_management_system) }
+
+    it "destroy when has no managers" do
+      expect(provider).to receive(:destroy)
+      provider.destroy_queue
+    end
+
+    it "call orchestrate_destroy its managers first" do
+      expect(manager).to receive(:enabled) { true }
```

Comment:
Don't do this, just `manager = FactoryGirl.create(:ext_management_system, :enabled => true)`

Reply:
done